service/tablet_allocator.cc

```diff
@@ -1007,8 +1124,100 @@ class load_balancer {
     }
 
     future<tablet_rack_list_colocation_plan> make_rack_list_colocation_plan(const migration_plan& mplan) {
+        lblogger.debug("In make_rack_list_colocation_plan");
+
         tablet_rack_list_colocation_plan plan;
-        co_return plan;
+        if (!ongoing_rack_list_colocation()) {
+            co_return plan;
+        }
+
+        const locator::topology& topo = _tm->get_topology();
+
+        auto migration_tablet_ids = co_await mplan.get_migration_tablet_ids();
+        colocation_sources_by_destination_rack dst_dc_rack_to_tablets = co_await find_required_rack_list_colocations(_db, _tm, _sys_ks,
+            _topology->paused_rf_change_requests, std::move(migration_tablet_ids));
+
+        node_load_map nodes;
+        topo.for_each_node([&] (const locator::node& node) {
+            if (node.get_state() == locator::node::state::normal && !node.is_excluded()) {
+                ensure_node(nodes, node.host_id(), false);
+            }
+        });
+
+        // Consider load that is already scheduled.
+        co_await consider_scheduled_load(nodes);
+
+        // Consider load that is about to be scheduled.
+        co_await consider_planned_load(nodes, mplan);
+
+        std::unordered_set<global_tablet_id> colocation_tablet_ids;
+        for (auto& [dc_rack, colocation_sources] : dst_dc_rack_to_tablets) {
+            node_load_map rack_nodes = nodes | std::views::filter([&] (const auto& host_load) {
```

Comment:
node_load is a heavy object, but we're copying it. Can we keep a vector of host_ids?